### DISCUSSION TITLE
Build solution with projects in Net 5/6 and Framework 4.8 (WPF)

### DISCUSSION THREAD

**Discussion by AelyPro** [cake-build/cake#3875] [Q&A] [answered]
Hey, I'm trying to migrate my company's custom versionning/generation tool to cake as it seems very promising and usefull.
I have a solution with project in Net 5 and 6 and one particular project doing backward compatibility in .Net Framework 4.8.

When I try to build the solution it run fine until it need to build this one project and then I have multiple error saying "X does not exist in the current context". It seem to run into trouble during the binding between the xaml and the code behind as all the error reference variable and function from the WPF framework ('InitializeComponent', 'IsVisibleChanged' , and component accessed from their x:Name).

To setup the pipeline I followed the guide "Setting Up A New Cake Frosting Project" but I'm having the same errors when trying to use the .Net Tool project.

```
public override void Run(BuildContext context)
    {
        context.DotNetBuild("../../CommonInteropNetFramework/CommonInteropNetFramework.csproj", new DotNetBuildSettings
        {
            Configuration = context.MsBuildConfiguration,
        });
    }
```

I also tried running DotNetMSBuild instead of DotNetBuild with the same errors.

Do I need to use the runner for .NET Framework on this specific project ?
Kinda new to all of this so I'm having trouble understanding what I'm doing wrong.

**Comment by devlead** [comment, marked as answer]
You could try using the MSBuild aliases which use full MSBuild that ships with Visual Studio / Build tools.

It should be capable of building both .NET and .NET Framework

Read more in docs:
https://cakebuild.net/dsl/msbuild/

**Comment by AelyPro** [comment]
I didn't realized they were another MsBuild function. Used it after importing "Cake.Common.Tools.MSBuild" and it worked fine.
Thanks !
